### DISCUSSION TITLE
android通过设置matrix缩放实现图片Layers实现宽高自适应出现异常位移问题

### DISCUSSION THREAD

**Discussion by WenxiaohuaEdward** [Tencent/libpag#1620] [Q&A] [answered]
PAG SDK 版本 ：com.tencent.tav:libpag:4.1.49
平台： Android 原生
预期实现效果：pag文件中imageLayers可根据实际图片的宽高来自适应
![image](https://github.com/Tencent/libpag/assets/10917249/9e402cd8-c6b8-43c0-a8be-8f8ea13e70b5)
实际效果：
![image](https://github.com/Tencent/libpag/assets/10917249/fb5d5397-3f55-42fc-afba-65e8a717e20f)
实际图片宽高是50*50， 图片Layers的宽高是48*48，通过上图代码设置matrix，Matrix{[1.0416666, 0.0, -1.0][0.0, 1.0416666, -1.0][0.0, 0.0, 1.0]}。   异常向下平移了一些位置，而我没有对matrix处理平移操作
![image](https://github.com/Tencent/libpag/assets/10917249/5664ed25-d55f-4641-bf38-8ad40aaf2ea1)

**Comment by WenxiaohuaEdward** [comment]
![image](https://github.com/Tencent/libpag/assets/10917249/8762f407-6ce2-4ada-902c-1f924378564d)
Matrix{[1.85, 0.0, -34.0][0.0, 1.5, -8.0][0.0, 0.0, 1.0]}.    图片实际宽高是148*48，imageLayer的宽高是80*32  

[效果.pag.zip](https://github.com/Tencent/libpag/files/12178712/pag.zip)

**Comment by kevingpqi123** [comment, marked as answer]
关于占位图图片填充，可以看下这篇文档， 这里面内置了几种填充模式： https://pag.art/docs/pag-fillmode.html

如果需要 setMatrix，可以参考下 iOS 端这个实现：https://github.com/libpag/pag-ios/blob/master/sample/pag-ios/ViewController/MoveAndRotateLayerViewController.swift
矩阵操作不是在原有 matrix 的基础上叠加设置，你的理解不是太对
